server/routes/api/student.js

```diff
@@ -0,0 +1,34 @@
+const express = require('express');
+const router = express.Router();
+var { Course, SeatingLayout } = require('../../dbSchemas/attendanceSchema');
+
+
+// allows updating a seating layout given a student id
+router.post('/updateStudentCourseView', (req, res) => {
```

Comment:
This API endpoint is supposed to allow updating the **seating_arrangement** (i.e. the student's positions), not the class layout. So the API should receive the `courseId`, and the 2d array of the new `seating_arrangement `so you should just need to find the course by id and overwrite the `seating_arrangement`.

Reply:
Yes, that makes more sense than what I did. Should be ok now.